src/focus-ring.js

```diff
@@ -1,67 +1,112 @@
-/* https://github.com/WICG/focusring */
+/* https://github.com/WICG/focus-ring */
 document.addEventListener('DOMContentLoaded', function() {
-    var hadKeyboardEvent = false,
-        keyboardModalityWhitelist = [ 'input:not([type])',
+    var hadKeyboardEvent = false;
+    var keyboardThrottleTimeoutID = 0;
+
+    var keyboardModalityWhitelist = [ 'input:not([type])',
                                       'input[type=text]',
                                       'input[type=number]',
                                       'input[type=date]',
                                       'input[type=time]',
                                       'input[type=datetime]',
                                       'textarea',
-                                      '[role=textbox]' ].join(','),
-        isHandlingKeyboardThrottle,
-        matcher = (function () {
-	    var el = document.body;
-	    if (el.matchesSelector)
-		return el.matchesSelector;
-	    if (el.webkitMatchesSelector)
-		return el.webkitMatchesSelector;
-	    if (el.mozMatchesSelector)
-		return el.mozMatchesSelector;
-	    if (el.msMatchesSelector)
-		return el.msMatchesSelector;
-	    console.error('Couldn\'t find any matchesSelector method on document.body.');
-	}()),
-	focusTriggersKeyboardModality = function (el) {
-	    var triggers = false;
-	    if (matcher) {
-		triggers = matcher.call(el, keyboardModalityWhitelist) && matcher.call(el, ':not([readonly]');
-	    }
-	    return triggers;
-	},
-        addFocusRingClass = function(el) {
-            if (el.classList.contains('focus-ring'))
-                return;
-            el.classList.add('focus-ring');
-            el.setAttribute('data-focus-ring-added', '');
-        },
-        removeFocusRingClass = function(el) {
-            if (!el.hasAttribute('data-focus-ring-added'))
-                return;
-            el.classList.remove('focus-ring');
-            el.removeAttribute('data-focus-ring-added')
-        };
+                                      '[role=textbox]' ].join(',');
+    var matcher = (function() {
```

Comment:
would you mind renaming this to `matches`? And adding a comment linking to the MDN post https://developer.mozilla.org/en-US/docs/Web/API/Element/matches

when I think of "matcher" my brain automatically assumes that this is already bound to a specific element. I have gone back and reread this chunk of code like 5 times and I still can't shake that feeling 😋